go-controller/pkg/ovn/pods.go

```diff
@@ -533,15 +535,10 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 		// namespace annotations to go through external egress router
 		if extIPs, err := getExternalIPsGRSNAT(oc.watchFactory, pod.Spec.NodeName); err != nil {
 			return err
-		} else if err = addOrUpdatePerPodGRSNAT(oc.nbClient, pod.Spec.NodeName, extIPs, podIfAddrs); err != nil {
+		} else if ops, err = oc.addOrUpdatePerPodGRSNATReturnOps(pod.Spec.NodeName, extIPs, podIfAddrs); err != nil {
```

Comment:
@tssurya I think Jaime's moved in the direction of having the "Ops" functions take an array of ops and return the appended ops. eg

`func (oc *Controller) addOrUpdatePerPodGRSNATReturnOps(nodeName string, extIPs, podIfAddrs []*net.IPNet) ([]ovsdb.Operation, error) {`

should become:

`func (oc *Controller) addOrUpdatePerPodGRSNATReturnOps(nodeName string, extIPs, podIfAddrs []*net.IPNet, ops []libovsdb.Operation) ([]ovsdb.Operation, error) {`

and then pass those ops into `CreateOrUpdateNATsOps`